# ErvSummary
<a name="top"></a>
This repository contains code scripts and sample data for analyzing germline mutation variability among different genetic contexts using extremely rare variants (ERVs).  

## Contents

### [1. Data Format](#format)  
### [2. Usage](#usage)
### [3. To do](#todo)



## 1. Data Format  
<a name="format"></a>

**ERV data**		A sample of ERV data (used for counting ERVs) can be found in `data/files_erv`;   
**Reference data**	A sample of reference data (used for counting reference motifs) can be found in `data/files_ref`;  
**Output data** 	A sample of the data generated by running `main.py` can be found in `data/output`.  
  

## 2. Usage  
<a name="usage"></a>  

### Download
Download the repository to local system.  

### Run the Program
Arguments:  
- `ervdir`: directory containing ERV data (the program reads all files in the directory)
- `refdir`: directory containing reference data (the program reads all files in the directory)
- `outputdir`: directory to save the output data  
- `nmer`: length of the sequence to be examined (for analyzing the sample ERV data, nmer should be integers from 1 to 5)  
- `center`: position of the variant in ERV data, index starts from 0. (the sample ERV data has the variant at the 5th of a 9-mer position, therefore center should take the value 4)  

Structure of the command:
`main.py -e <ervdir> -r <refdir> -o <outputdir> -n <nmer> -c <center>`  

Example:   
`python code/main.py -e data/files_erv -r data/files_ref -o data -n 5 -c 4`  
The command above counts 5-mer patterns from 9mer ERV data
  

## 3. To do 
<a name="todo"></a>  
- integrate plotting function (`code/plotGen.R`)
- adapt to original vcf file in order to examine longer sequences

  
[back to top](#top)
